lib/galaxy/webapps/galaxy/api/workflows.py

```diff
@@ -1005,6 +1007,46 @@ def invocation_step(self, trans, invocation_id, step_id, **kwd):
         )
         return self.__encode_invocation_step(trans, invocation_step)
 
+    def _generate_aws_estimate(self, trans, invocation_id, **kwd):
+        """
+        build AWS cloud estimate
+        """
+        decoded_workflow_invocation_id = self.decode_id(invocation_id)
+        workflow_invocation = self.workflow_manager.get_invocation(trans, decoded_workflow_invocation_id)
+        history = workflow_invocation.history
+
+        aws_dict, metrics, ds_metrics = {}, {}, {}
+        h_contents = self.history_contents_manager.contained(history)
```

Comment:
I would suggest not iterating over the history contents here, this may contain many more datasets that are not part of the workflow invocation. You can get size and other hda attributes from the job inputs and outputs

Reply:
If you build this dictionary in the for loop below you'd also address @bgruening's comment